01_classify_stratified/nbconverted/plot_cancer_type_results.py

```diff
@@ -283,3 +285,136 @@ def label_points(x, y, gene, sig, ax):
 plt.title('Cancer type confusion matrix, methylation data', size=14, pad=14)
 plt.tight_layout()
 
+
+# How to read the confusion matrices:
+# 
+# * Remember, we're doing one-vs-rest classification of the cancer types. The cancer type on the y-axis is what the model is trained to predict (i.e. the positive label in the training set), and the cancer type on the x-axis is what we're evaluating on (i.e. the true label in the test set).
+# 
+# * A high off-diagonal value in the heatmap indicates that a model trained to distinguish the y-axis cancer type from all the others often "confuses" it for the x-axis cancer type (i.e. they have similar gene expression profiles). Diagonal values indicate the accuracy of the model on the target cancer type (these values should be relatively high).
+# 
+# In the resulting heatmaps, we can observe a few things:
+# 
+# * The models are accurate overall (high on-diagonal values and mostly low off-diagonal values), and the "inaccuracies" (high off-diagonal values) we see aren't surprising based on our knowledge of relationships between cancer types.
+# 
+# * Cancer type predictions using methylation data are noisier and not quite as accurate compared to gene expression data.
+
+# ## Hierarchical clustering analysis
+# 
+# We want to see if these confusion matrices lead to reasonable "cancer groups" (i.e. are cancer types that we would expect to be similar close by in the "similarity matrix" given by model confusion?). To do this, we'll just do a super simple agglomerative hierarchical clustering.
```

Comment:
Think there might be a typo in the parenthesis

Reply:
No typo (that I know of), but the wording here is def awkward. I'll rephrase to hopefully be a bit clearer.